codegen/src/next/mod.rs

```diff
@@ -237,13 +279,46 @@ impl ToTokens for App {
     fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
         let Self { endpoints } = self;
 
+        let mut endpoint_chains = endpoints
+            .iter()
+            .fold(HashMap::new(), |mut chain, endpoint| {
+                let entry = chain
+                    .entry(&endpoint.route)
+                    .or_insert_with(Vec::<Handler>::new);
+
+                let method = endpoint.method.clone();
+                let function = endpoint.function.clone();
+
+                if entry.iter().any(|handler| handler.method == method) {
+                    emit_error!(
+                        method,
+                        "only one method of each type is allowed per route";
+                        hint = format!("Remove one of the {} methods on the \"{}\" route.", method, endpoint.route.value())
+                    );
+                } else {
+                    entry.push(Handler { method, function });
+                }
+
+                chain
+            })
+            .into_iter()
+            .map(|(key, value)| EndpointChain {
+                route: key,
+                handlers: value,
+            })
+            .collect::<Vec<EndpointChain>>();
+
+        // Sort endpoints by route alphabetically to ensure test consistency
+        endpoint_chains.sort_by(|a, b| a.route.value().cmp(&b.route.value()));
```

Comment:
Swapping the `HashMap` for a `BTreeMap` should remove the need for this sort

Reply:
I tried this, but unfortunately `LitStr` doesn't implement `Ord`. On second thought I could implement `Ord` for it, and compare the `.value()` in the `cmp` methods.

Reply:
Okay, this is simpler then :smile: